### DISCUSSION TITLE
Welcome to autopahe Discussions!

### DISCUSSION THREAD

**Discussion by haxsysgit** [haxsysgit/autopahe#6] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by roso4991** [comment]
Is there a way to filter and download the dubbed version of episodes?

**Comment by haxsysgit** [comment]
Well I could implement that in the next version , but just so you know,
there's barely a dubbed version of most animes in Animepahe

**Reply by roso4991** [reply]
That would be great for when it does exist though. Thanks!

**Comment by haxsysgit** [comment]
I'll probably do it tomorrow, would be used with a --dub argument

**Comment by terriblesarcasm** [comment]
It looks like the episode caches are based on page 1 of the anime, when getting to episodes on page 2 (episode 31+) it immediately dies and doesn't attempt to get the files.

I attempted to get the respective page in the download function but it never makes it passed the cache to get to it:
`page_number = (arg + 29) // 30`
`stream_page_url = f'https://animepahe.com/play/{session_id}/{episode_session}?page={page_number}'`

**Reply by haxsysgit** [reply]
Hmmm, I'll check it out , maybe because all the anime I watch is usually 12 eps , I ignored it lol

**Reply by haxsysgit** [reply]
it's fixed already , did it that way due to optimization and performance issues

**Comment by Nova1422** [comment]
Only tool which worked for me!

**Reply by haxsysgit** [reply]
so happy to hear that , if you have any suggestions or idea to improve this tool , feel free to discuss here

**Reply by Nova1422** [reply]
id suggest something to make editing the default config easier i suppose, unless its a me issue, because personally, i dont seem to understand how it works, i believe you have to `autopahe --write-config` and edit the generated config file, which itll pick up directly right?

**Reply by Nova1422** [reply]
I set it up with variables previously and mid way it stopped working so I started going all over the place

Here's what I discovered, the variable setup works atleast, but for some reason, when I change the quality to something other than 720, the config.ini stops loading entirely, if the quality is 720, it follows the download directly and sorting directions

**Reply by haxsysgit** [reply]
oh no it isn't you , i just went thru the config implementation and saw the problem, can you send a screenshot of what it shows in your terminal

**Reply by haxsysgit** [reply]
also i'm thinking of using docker to fix the "but it works on my system" problem ,held off on doing that cos there werent that many users , what do you think?

**Reply by Nova1422** [reply]
Seem like you already got an update up, also my knowledge is limited in this aspects, but from a mild research, if you have time, using docker might be a good idea, I'm on windows, sure the program works fine, but maybe it doens't do everything it could, one thing I can recall would be ""clear" not recognized as an internal or external command, operable program or batch file." Which pops up after every autopahe command

**Reply by Nova1422** [reply]
I'll test out the update in 24hrs, and do you still need the screenshots? If so, about what?

**Reply by haxsysgit** [reply]
Also this may sound funny but i only use the tool for downloading(streaming sometimes) and sorting the downloaded anime 😂  other functions are just convenience functions I added when I was bored or had free time, sorry for the sloppy software lol

**Reply by haxsysgit** [reply]
Also Yeah I kinda still need the screenshots about the problem, my PC is actually dual boot Linux but I mostly use Linux , I didn't test it very well on Windows , but I built it with windows in mind

**Reply by Nova1422** [reply]
<img width="1920" height="1080" alt="Screenshot (817)" src="https://github.com/user-attachments/assets/cd361ec2-3120-4d1f-9abf-c6fdab2b8b76" />
<img width="1920" height="1080" alt="Screenshot (816)" src="https://github.com/user-attachments/assets/7997dcfd-cfab-41c0-969d-58b0cf9a5baf" />
<img width="1920" height="1080" alt="Screenshot (815)" src="https://github.com/user-attachments/assets/f0b3c425-f027-45e1-830f-0d1f2d4228d8" />
<img width="1920" height="1080" alt="Screenshot (814)" src="https://github.com/user-attachments/assets/68d444a6-5a02-47c5-a8c9-b65369af75bf" />
i dunno what exact issues youre reffering to, but here are some struggles of mine trying to update the downloader in the meanwhile

as you can see, cmd refused to pick it up, and kept defaulting to pypi installation

**Reply by Nova1422** [reply]
there are things i encountered trying to get the downloader working by git clone, i didnt understand how to run them so i ended up with pip install git+https://github.com/haxsysgit/autopahe.git, it seems to do the update

**Reply by Nova1422** [reply]
also, --write-config does this
<img width="1920" height="1080" alt="Screenshot (818)" src="https://github.com/user-attachments/assets/b40841fb-53de-4262-99bf-c3856a6691ab" />

**Reply by Nova1422** [reply]
C:\Users\DELL>autopahe config edit
ERROR: Failed to open editor: [WinError 2] The system cannot find the file specified

**Reply by Nova1422** [reply]
All Animepahe files have been renamed

Directory Cleaned -> E:\
Files Current path -> E:\AnimePahe

i want to make it so the edited files are stored in E dir itself, how can i edit that? also, the files werent moved

edit: they werent renamed either

config.ini for context 

Optional: automatically sort after downloads (not enabled by default)
sort_on_complete = true

Optional: preferred sort mode when enabled: all | rename | organize
sort_mode = all

Optional: sort path override (leave empty to use download_dir or OS default)
sort_path = E:\

**Reply by Nova1422** [reply]
sorry if im asking too many questions lol

**Reply by Nova1422** [reply]
autopahe --collection organize also seems to make 2 folder based on anime name, one in download dir and one in E dir (where the eps are) and proceeds to move the files from E dir and sorts them in download dir, dont ask how that works

**Reply by haxsysgit** [reply]
damn lxml doesn't work on windows? thats a core module for autopahe , i'll have to go to windows and test it

**Reply by haxsysgit** [reply]
also no , you arent asking lots of question , always happy to see your engagement, i guess i'll have to change my environment to full time windows for a while

**Reply by Nova1422** [reply]
editor = os.environ.get('EDITOR') or 'vi'

might to change that to include notepad or windows, it does this

C:\Users\DELL>autopahe config edit
ERROR: Failed to open editor: [WinError 2] The system cannot find the file specified

**Reply by Nova1422** [reply]
Also, autopahe --setup doesn't work on windows, atleast for pip, install.py not found it says, I used pip for first half and proceeded with uv as anything else didn't work and I didn't wanted to clone the repo, I think the problem still persisted even after I had cloned the repo but I can't confirm it atp

And remember the prev config file error I mentioned?

The variable config.ini setup is still very moody, the config shows up in autopahe config show/validate, but it may or may not load, it's most stable at 720p, any change in that and it defaults to defaults

**Reply by Nova1422** [reply]
If the file is in .config it seems to work fine tho

Also are all files set to sort after completion meant to go in animepahe/ ? 


Edit: not much left to point out, it seems, concludes my rambling until I can eventually get time to watch animes again

**Reply by haxsysgit** [reply]
is everything alright now?

**Reply by Nova1422** [reply]
I didn't have notifs on for new commits, cant have my laptop for a while I'll lyk in a day or two

**Comment by haxsysgit** [comment]
Forgot vi isn't the default option in Windows, thanks a lot for the
suggestion, always look forward to this every morning

**Reply by Nova1422** [reply]
Glad to hear that!

**Comment by haxsysgit** [comment]
Yep I made it that way , because I assumed people get animes from different
sources, so it's best to group which is from Animepahe

**Comment by kvrkarthik** [comment]
can i put -st 1-12 -p 1080? Like multi-episode streaming? Is that possible? I tried but failed to do it.

**Reply by haxsysgit** [reply]
yes you can but make sure you have the supported players  or just paste the error message here lemme see 
You must include -s and -i in the same command because selection isn’t persisted between runs.
-p only accepts 360|480|720|1080|best|worst. If some episodes don’t have >=720p, -p 1080 can fail; try -p best or -p 720.

**Reply by kvrkarthik** [reply]
Like, my problem is I am trying to multi-stream and my anime has 1080p available and now I put 1-3(for eg.) and then it plays first episode on vlc as I put "--player vlc", then after 1st episode i tried playing next episode, but I couldn't do it

**Reply by haxsysgit** [reply]
okay yeah i just made a quick fix for that , i realized it wasn't caching properly , lemme push it so you can see the changes

**Comment by Dannyboy1506** [comment]
Please can you add an update that prints the direct download link in the terminal so that one could use any browser of their choice to download

**Reply by Nova1422** [reply]
Does -l not work?

**Reply by haxsysgit** [reply]
Exactly lol, I was was very sure that I added that feature long time ago

Use -l or --link

**Reply by Nova1422** [reply]
Oh btw, the links outdated

**Reply by Nova1422** [reply]
and the install.py not found text is still prevalent, did i overlook or forget smthng?


i mean, the system does kinda load but idk

**Reply by Dannyboy1506** [reply]
I didn't know there was a -l flag have you tried using the script recently I'm getting status code 403 at kwik

**Comment by haxsysgit** [comment]
Quick update: v3.6.0 is out now.

  The `403 at kwik` issue is real. Kwik is now blocking automated downloads with security verification, so I’m not going to pretend AutoPahe can reliably bypass that.

  Starting from v3.6.0, downloads are now browser-required:

  - `-d` / `--single_download` now prepares the browser download link and shows the expected folder + suggested filename.
  - AutoPahe only marks the episode as downloaded after it verifies a real local video file.
  - `-l` / `--link` still exists if you only want the link and do not want records updated.

  Update with:

  ```bash
 pip install -U autopahe
```
with uv:

```bash
uv tool upgrade autopahe
```
  or, if you already have the new updater available:

```bash
autopahe --update
```

  I also fixed the stale AnimePahe domain links. The code now uses the current .pw, .com, and .org mirrors instead of the old .ru / .si links.

  For the install.py not found report: please update to 3.6.0 first and try again. If it still happens, send the exact OS, install method, and command you ran so I can isolate that
  separately.

**Reply by kvrkarthik** [reply]
My issue is:
I ran a command to stream (or maybe wish to download) and it ran searching for a long time, maybe like a minute or two, and then it finally says that no such anime is found while searching and exited. 
I do not know why. So please try to fix that.
